cmd/finish_image_build.go

```diff
@@ -93,13 +97,44 @@ func (f *FinishImageBuild) SetFlags(flags *flag.FlagSet) {
 	flags.BoolVar(&f.inheritLabels, "inherit-labels", false, "Indicates if the result image should inherit labels "+
 		"from the source image. Labels specified through the '-labels' flag take precedence over inherited "+
 		"labels.")
+	flags.StringVar(&f.oemSize, "oem-size", "", "Size of the new OEM partition, "+
+		"can be a number with unit like 10G, 10M, 10K or 10B, "+
+		"or without unit indicating the number of 512B sectors.")
 	flags.IntVar(&f.diskSize, "disk-size-gb", 0, "The disk size to use when creating the image in GB. Value of '0' "+
 		"indicates the default size.")
 	flags.DurationVar(&f.timeout, "timeout", time.Hour, "Timeout value of the image build process. Must be formatted "+
 		"according to Golang's time.Duration string format.")
 }
 
 func (f *FinishImageBuild) validate() error {
+	// The default size of a COS image (imgSize) is 10GB. And the OEM partition size should be doubled to
+	// store the hash tree of dm-verity. If the OEM partition is to be extended,
+	// the following must be true: disk-size - oem-size x 2 >= imgSize.
+	// Since we allow user input like "500M", and the "resize-disk" API can only take GB as input,
+	// the oem-size is rounded up to GB to make sure there is enough space.
+	// Extra space will be taken by the stateful partition.
+	const imgSize = 10
+	if f.oemSize != "" {
+		oemSizeBytes, err := partutil.ConvertSizeToBytes(f.oemSize)
+		if err != nil {
+			return fmt.Errorf("invalid format of oem-size: %q, error msg:(%v)", f.oemSize, err)
+		}
+		f.oemFSSize4K = oemSizeBytes >> 12
+		// double the oem size.
+		oemSizeBytes <<= 1
```

Comment:
Let's leave the dm-verity stuff out of this PR. We can have a separate PR for incorporating all of the dm-verity work.

Reply:
Discussed offline, we'll keep this in for now and improve in follow-up PR.